### PR TITLE
Add ci action to publish changelog items to changelog branch

### PR DESCRIPTION
**Pull request by metal-messiah**
### Overview
This PR adds a CI action to add a changelog item to the [changelog branch](https://github.com/newrelic/newrelic-browser-agent/tree/changelog) when PRs are merged to main.

The structure of the final merge commit message is what will determine the output of the changelog item.

The `main commit message` will serve as the item "heading" while the `extended description` will serve as the item "body".
The changelog item will also create a link to the PR itself, and you can also include helpful external links in the extended description if needed.

### Related Issue(s)
This is part of an Innovation Project

### CODE REVIEW

File: .github/workflows/add-changelog-item.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  add-changelog-item:
+    runs-on: ubuntu-latest
+    container: node:14
```
Comment (patrickhousley) [inline .github/workflows/add-changelog-item.yml:11]: Can you take a look at how the ci workflow is setup and mimic that. That way we can use the latest version of node.

File: .github/workflows/add-changelog-item.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: changelog
```
Comment (patrickhousley) [inline .github/workflows/add-changelog-item.yml:15]: Looks like the branch has node_modules committed. Just heads up.

Comment (metal-messiah) [reply .github/workflows/add-changelog-item.yml:15]: TY

Comment (metal-messiah) [reply .github/workflows/add-changelog-item.yml:21]: removed in the branch, thanks

**Review by patrickhousley — APPROVED** [review]

File: .github/workflows/add-changelog-item.yml (hunk 1 above)
Comment (patrickhousley) [reply .github/workflows/add-changelog-item.yml]: Looks good.

**Comment by metal-messiah** [comment]
Considering an alternative solution in tandem with @bjfield to support agent-metadata initiative

**Pull request closed without merging**